v2/pubsub-to-redis/src/main/java/com/google/cloud/teleport/v2/templates/PubSubToRedis.java

```diff
@@ -0,0 +1,350 @@
+
+    @TemplateParameter.Integer(
+        order = 7,
+        optional = true,
+        description = "Redis connection timeout in ms",
+        helpText = "Redis connection timeout in ms.",
+        example = "2000")
+    @Default.Integer(2000)
+    int getConnectionTimeout();
+
+    void setConnectionTimeout(int timeout);
+
+    @TemplateParameter.Long(
+        order = 8,
+        optional = true,
+        description = "Hash key expiration time in sec (ttl)",
+        helpText =
+            "Key expiration time in sec (ttl, default for HASH_SINK is -1 i.e. no expiration)")
+    @Default.Long(-1L)
+    Long getTtl();
+
+    void setTtl(Long ttl);
+
+    /*
+    @TemplateParameter.Long(
+            order = 9,
+            optional = true,
+            description = "Duration in seconds of a pane in a Global window",
+            helpText = "Windowing pipeline with sessions window")
+    @Default.Long(5)
+    Long getWindowDuration();
+
+    void setWindowDuration(Long windowDuration);
+     */
+  }
+
+  /** Allowed list of sink types. */
+  public enum RedisSinkType {
+    HASH_SINK,
+    LOGGING_SINK,
+    STREAMS_SINK,
+    STRING_SINK
+  }
+
+  /**
+   * Main entry point for executing the pipeline.
+   *
+   * @param args The command-line arguments to the pipeline.
+   */
+  public static void main(String[] args) {
+    UncaughtExceptionLogger.register();
+
+    // Parse the user options passed from the command-line.
+    PubSubToRedisOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(PubSubToRedisOptions.class);
+    run(options);
+  }
+
+  /**
+   * Runs the pipeline with the supplied options.
+   *
+   * @param options The execution parameters to the pipeline.
+   * @return The result of the pipeline execution.
+   */
+  public static PipelineResult run(PubSubToRedisOptions options) {
+
+    // Create the pipeline
+    Pipeline pipeline = Pipeline.create(options);
+
+    PCollection<PubsubMessage> input;
+
+    RedisConnectionConfiguration redisConnectionConfiguration =
+        RedisConnectionConfiguration.create()
+            .withHost(options.getRedisHost())
+            .withPort(options.getRedisPort())
+            .withAuth(options.getRedisPassword())
+            .withTimeout(options.getConnectionTimeout())
+            .withSSL(options.getSslEnabled());
+
+    /*
+     * Steps: 1) Read PubSubMessage with attributes and messageId from input PubSub subscription.
+     *        2) Extract PubSubMessage message to PCollection<String>.
+     *        3) Transform PCollection<String> to PCollection<KV<String, String>> so it can be consumed by RedisIO
+     *        4) Write to Redis using SET
+     *
+     */
+
+    LOG.info(
+        "Starting PubSub-To-Redis Pipeline. Reading from subscription: {}",
+        options.getInputSubscription());
+
+    input =
+        pipeline.apply(
+            "Read PubSub Events",
+            MessageTransformation.readFromPubSub(options.getInputSubscription()));
```

Comment:
Do we plan to support UDFs here? It appears so from the docs, but I don't see the transforms

Reply:
I believe I have removed all traces of UDFs for now, so you shouldn't see anything related to UDF transformations. We can add UDF in the next round.